--- mpg123/Makefile.orig	Tue Jan 23 19:00:46 2001
+++ mpg123/Makefile	Tue Feb  4 23:03:44 2003
@@ -152,7 +152,7 @@
 linux-devel:
 	$(MAKE) OBJECTS='decode_i386.o dct64_i386.o audio_oss.o' \
         CC=gcc LDFLAGS= \
-        CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DLINUX -Wall -g -m486 \
+        CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DREAL_IS_FLOAT -DLINUX -Wall \
 		-DOSS -funroll-all-loops \
 		-finline-functions -ffast-math' \
         mpg123-make
@@ -160,7 +160,7 @@
 linux-profile:
 	$(MAKE) OBJECTS='decode_i386.o dct64_i386.o audio_oss.o' \
         CC=gcc LDFLAGS='-pg' \
-        CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DLINUX -Wall -pg -m486 \
+        CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DREAL_IS_FLOAT -DLINUX -Wall -pg \
 		-DOSS -funroll-all-loops \
 		-finline-functions -ffast-math' \
         mpg123-make
@@ -169,10 +169,10 @@
 	$(MAKE) CC=gcc LDFLAGS= \
 		OBJECTS='decode_i386.o dct64_i386.o decode_i586.o \
 			audio_oss.o term.o' \
-		CFLAGS='$(CFLAGS) -DI386_ASSEM -DPENTIUM_OPT -DREAL_IS_FLOAT -DLINUX \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DI386_ASSEM -DPENTIUM_OPT -DREAL_IS_FLOAT -DLINUX \
 			-DOSS -DTERM_CONTROL\
-			-Wall -O2 -m486 \
-			-fomit-frame-pointer -funroll-all-loops \
+			-Wall \
+			-funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
 
@@ -180,10 +180,10 @@
 	$(MAKE) CC=gcc LDFLAGS= \
 		OBJECTS='decode_i386.o dct64_i386.o decode_i586.o \
 			audio_oss.o term.o' \
-		CFLAGS='$(CFLAGS) -DI386_ASSEM -DPENTIUM_OPT -DREAL_IS_FLOAT -DLINUX \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DI386_ASSEM -DPENTIUM_OPT -DREAL_IS_FLOAT -DLINUX \
 			-DOSS -DTERM_CONTROL\
-			-Wall -O2 -mpentium \
-			-fomit-frame-pointer -funroll-all-loops \
+			-Wall \
+			-funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
 
@@ -191,10 +191,10 @@
 	$(MAKE) CC=gcc LDFLAGS= \
 		OBJECTS='decode_i386.o dct64_MMX.o tabinit_MMX.o decode_MMX.o \
 			audio_oss.o term.o' \
-		CFLAGS='-DUSE_MMX -DI386_ASSEM -DPENTIUM_OPT -DREAL_IS_FLOAT \
+		CFLAGS='$(OPT_FLAGS) -DUSE_MMX -DI386_ASSEM -DPENTIUM_OPT -DREAL_IS_FLOAT \
 			-DLINUX -DOSS -DTERM_CONTROL\
-			-Wall -O2 -m486 \
-			-fomit-frame-pointer -funroll-all-loops \
+			-Wall \
+			-funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
 
@@ -203,10 +203,10 @@
 		OBJECTS='decode_i386.o decode_3dnow.o dct64_3dnow.o \
 			dct64_i386.o dct36_3dnow.o getcpuflags.o \
 			equalizer_3dnow.o decode_i586.o audio_oss.o term.o' \
-		CFLAGS='$(CFLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DLINUX \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DLINUX \
 			-DUSE_3DNOW -DOSS -DTERM_CONTROL\
-			-Wall -O2 -m486 \
-			-fomit-frame-pointer -funroll-all-loops \
+			-Wall \
+			-funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
 
@@ -216,10 +216,10 @@
 		OBJECTS='decode_i386.o dct64_i386.o decode_i586.o \
 			decode_i486.o audio_oss.o term.o \
 			dct64_i486-a.o dct64_i486-b.o ' \
-		CFLAGS='$(CFLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DI486_OPT -DLINUX \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DI486_OPT -DLINUX \
 			-DOSS -DTERM_CONTROL\
-			-Wall -O2 -m486 \
-			-fomit-frame-pointer -funroll-all-loops \
+			-Wall \
+			-funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
 
@@ -228,12 +228,11 @@
 		AUDIO_LIB='-lesd -laudiofile' \
 		OBJECTS='decode_i386.o dct64_i386.o decode_i586.o \
 			audio_esd.o' \
-		CFLAGS='$(CFLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DLINUX \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DLINUX \
 			-DOSS -DUSE_ESD \
-			-Wall  -O2 -m486 \
-			-fomit-frame-pointer -funroll-all-loops \
-			-finline-functions -ffast-math \
-			$(RPM_OPT_FLAGS)' \
+			-Wall \
+			-funroll-all-loops \
+			-finline-functions -ffast-math' \
 		mpg123-make
 
 linux-alsa:
@@ -241,12 +240,36 @@
 		AUDIO_LIB='-lasound' \
 		OBJECTS='decode_i386.o dct64_i386.o decode_i586.o \
 			audio_alsa.o term.o' \
-		CFLAGS='$(CFLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DLINUX \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DLINUX \
 			-DALSA -DTERM_CONTROL\
-			-Wall  -O2 -m486 \
-			-fomit-frame-pointer -funroll-all-loops \
-			-finline-functions -ffast-math \
-			$(RPM_OPT_FLAGS)' \
+			-Wall \
+			-funroll-all-loops \
+			-finline-functions -ffast-math' \
+		mpg123-make
+
+linux-i486-esd:
+	$(MAKE) CC=gcc LDFLAGS= \
+		AUDIO_LIB='-lesd -laudiofile' \
+		OBJECTS='decode_i386.o dct64_i386.o decode_i586.o \
+			decode_i486.o audio_esd.o \
+			dct64_i486-a.o dct64_i486-b.o ' \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DI486_OPT -DLINUX \
+			-DOSS -DUSE_ESD \
+			-Wall \
+			-funroll-all-loops \
+			-finline-functions -ffast-math' \
+		mpg123-make
+
+linux-mmx-esd:
+	$(MAKE) CC=gcc LDFLAGS= \
+		AUDIO_LIB='-lesd -laudiofile' \
+		OBJECTS='decode_i386.o dct64_MMX.o tabinit_MMX.o decode_MMX.o \
+			audio_esd.o' \
+		CFLAGS='$(OPT_FLAGS) -DUSE_MMX -DI386_ASSEM -DPENTIUM_OPT -DREAL_IS_FLOAT \
+			-DLINUX -DOSS -DUSE_ESD \
+			-Wall \
+			-funroll-all-loops \
+			-finline-functions -ffast-math' \
 		mpg123-make
 
 linux-3dnow-alsa:
@@ -255,10 +278,10 @@
 		OBJECTS='decode_i386.o decode_3dnow.o dct64_3dnow.o \
 			dct64_i386.o dct36_3dnow.o getcpuflags.o \
 			equalizer_3dnow.o decode_i586.o audio_alsa.o term.o' \
-		CFLAGS='-DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DLINUX \
+		CFLAGS='$(OPT_FLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DLINUX \
 			-DUSE_3DNOW -DALSA -DTERM_CONTROL\
-			-Wall -O2 -m486 \
-			-fomit-frame-pointer -funroll-all-loops \
+			-Wall \
+			-funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
 
@@ -268,10 +291,10 @@
 		OBJECTS='decode_i386.o decode_3dnow.o dct64_3dnow.o \
 			dct64_i386.o dct36_3dnow.o getcpuflags.o \
 			equalizer_3dnow.o decode_i586.o audio_esd.o' \
-		CFLAGS='-DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DLINUX \
+		CFLAGS='$(OPT_FLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DLINUX \
 			-DUSE_3DNOW -DUSE_ESD \
-			-Wall -O2 -m486 \
-			-fomit-frame-pointer -funroll-all-loops \
+			-Wall \
+			-funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
 
@@ -279,43 +302,39 @@
 	$(MAKE) CC=gcc LDFLAGS= \
 		AUDIO_LIB='-lasound' \
 		OBJECTS='decode.o dct64.o audio_alsa.o term.o' \
-		CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DLINUX -DALSA \
-			-DTERM_CONTROL -Wall  -O2 \
-			-fomit-frame-pointer -funroll-all-loops \
-			-finline-functions -ffast-math \
-			$(RPM_OPT_FLAGS)' \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DREAL_IS_FLOAT -DLINUX -DALSA \
+			-DTERM_CONTROL -Wall \
+			-funroll-all-loops \
+			-finline-functions -ffast-math' \
 		mpg123-make
 
 linux-alpha:
 	$(MAKE) CC=gcc LDFLAGS= OBJECTS='decode.o dct64.o audio_oss.o' \
-		CFLAGS='$(CFLAGS) -DLINUX -DOSS -Wall -O2 \
-			-fomit-frame-pointer -funroll-all-loops \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DLINUX -DOSS -Wall \
+			-funroll-all-loops \
 			-finline-functions -ffast-math \
-			-Wall -O6 -DUSE_MMAP \
-			$(RPM_OPT_FLAGS)' \
+			-Wall -DUSE_MMAP' \
 		mpg123-make
 
 linux-alpha-alsa:
 	$(MAKE) CC=gcc LDFLAGS= \
 		AUDIO_LIB='-lasound' \
 		OBJECTS='decode.o dct64.o audio_alsa.o term.o' \
-		CFLAGS='-DLINUX \
+		CFLAGS='$(OPT_FLAGS) -DLINUX \
 			-DALSA -DTERM_CONTROL\
-			-DUSE_MMAP  -O6 \
-			-fomit-frame-pointer -funroll-all-loops \
-			-finline-functions -ffast-math \
-			$(RPM_OPT_FLAGS)' \
+			-DUSE_MMAP \
+			-funroll-all-loops \
+			-finline-functions -ffast-math' \
 		mpg123-make
 
 linux-alpha-esd:
 	$(MAKE) CC=gcc LDFLAGS= \
 		AUDIO_LIB='-lesd -laudiofile' \
 		OBJECTS='decode.o dct64.o audio_esd.o' \
-		CFLAGS='$(CFLAGS) -DLINUX -DOSS -DUSE_ESD -Wall -O2 \
-			-fomit-frame-pointer -funroll-all-loops \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DLINUX -DOSS -DUSE_ESD -Wall \
+			-funroll-all-loops \
 			-finline-functions -ffast-math \
-			-Wall -O6 -DUSE_MMAP \
-			$(RPM_OPT_FLAGS)' \
+			-Wall -DUSE_MMAP' \
 		mpg123-make
 
 #linux-ppc:
@@ -340,9 +359,9 @@
 linux-ppc:
 	$(MAKE) CC=gcc  LDFLAGS= \
 		OBJECTS='decode.o dct64.o audio_oss.o' \
-		CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DLINUX -Wall -O2 -mcpu=ppc \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DREAL_IS_FLOAT -DLINUX -Wall \
 			-DOSS \
-			-fomit-frame-pointer -funroll-all-loops \
+			-funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
 
@@ -350,17 +369,17 @@
 	$(MAKE) CC=gcc  LDFLAGS= \
 		AUDIO_LIB='-lesd -laudiofile' \
 		OBJECTS='decode.o dct64.o audio_esd.o' \
-		CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DLINUX -Wall -O2 -mcpu=ppc \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DREAL_IS_FLOAT -DLINUX -Wall \
 			-DOSS -DUSE_ESD \
-			-fomit-frame-pointer -funroll-all-loops \
+			-funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
 
 linux-sparc:
 	$(MAKE) CC=gcc  LDFLAGS= \
 		OBJECTS='decode.o dct64.o audio_sun.o' \
-		CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DUSE_MMAP -DSPARCLINUX -Wall -O2 \
-			-fomit-frame-pointer -funroll-all-loops \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DREAL_IS_FLOAT -DUSE_MMAP -DSPARCLINUX -Wall \
+			-funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
 
@@ -368,34 +387,34 @@
 	$(MAKE) CC=gcc  LDFLAGS= \
 		AUDIO_LIB='-lesd -laudiofile' \
 		OBJECTS='decode.o dct64.o audio_esd.o' \
-		CFLAGS='-DREAL_IS_FLOAT -DUSE_MMAP -DOSS -DUSE_ESD -DSPARCLINUX -Wall -O2 \
-			-fomit-frame-pointer -funroll-all-loops \
-			-finline-functions -ffast-math \
+		CFLAGS='$(OPT_FLAGS) -DREAL_IS_FLOAT -DUSE_MMAP -DOSS -DUSE_ESD -DSPARCLINUX -Wall \
+			-funroll-all-loops \
+			-finline-functions -ffast-math' \
 		mpg123-make
 
 
 linux-armv4l:
 	$(MAKE) CC=gcc LDFLAGS= OBJECTS='decode.o dct64.o audio_oss.o' \
-		CFLAGS='-DLINUX -DOSS -Wall -O2 \
-			-fomit-frame-pointer -funroll-all-loops \
+		CFLAGS='$(OPT_FLAGS) -DLINUX -DOSS -Wall \
+			-funroll-all-loops \
 			-finline-functions -ffast-math \
-			-Wall -O6 -DUSE_MMAP \
+			-Wall -DUSE_MMAP \
                 mpg123-make
 
 linux-arm:
 	$(MAKE) CC=gcc LDFLAGS= OBJECTS='decode.o dct64.o audio_oss.o' \
-		CFLAGS='-DREAL_IS_FIXED -DLINUX \
-			-DOSS -Wall -O6 -march=armv4 \
-			-fomit-frame-pointer -funroll-all-loops \
+		CFLAGS='$(OPT_FLAGS) -DREAL_IS_FIXED -DLINUX \
+			-DOSS -Wall \
+			-funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
  
 
 linux-m68k:
 	$(MAKE) CC=gcc LDFLAGS= OBJECTS='decode.o dct64.o audio_oss.o' \
-		CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DLINUX \
-			-DOSS -DOSS_BIG_ENDIAN -Wall -O2 -m68040 \
-			-fomit-frame-pointer -funroll-loops \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DREAL_IS_FLOAT -DLINUX \
+			-DOSS -DOSS_BIG_ENDIAN -Wall \
+			-funroll-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
 
@@ -415,9 +434,9 @@
 	$(MAKE) CC=gcc LDFLAGS= \
 		OBJECTS='decode_i386.o dct64_i386.o decode_i586.o \
 			control_sajber.o control_tk3play.o audio_oss.o' \
-		CFLAGS='$(CFLAGS) -DFRONTEND -DOSS -DI386_ASSEM -DREAL_IS_FLOAT \
-			-DPENTIUM_OPT -DLINUX -Wall -O2 -m486 \
-			-fomit-frame-pointer -funroll-all-loops \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -DFRONTEND -DOSS -DI386_ASSEM -DREAL_IS_FLOAT \
+			-DPENTIUM_OPT -DLINUX -Wall \
+			-funroll-all-loops \
 			-finline-functions -ffast-math' \
 		$(FRONTEND)
 
@@ -425,10 +444,10 @@
 	$(MAKE) CC=gcc LDFLAGS='-L/usr/X11R6/lib' \
 		AUDIO_LIB='-laudio -lXau' \
 		OBJECTS='decode_i386.o dct64_i386.o audio_nas.o' \
-		CFLAGS='$(CFLAGS) -I/usr/X11R6/include \
+		CFLAGS='$(CFLAGS) $(OPT_FLAGS) -I/usr/X11R6/include \
 			-DI386_ASSEM -DREAL_IS_FLOAT -DLINUX -DNAS \
-			-Wall -O2 -m486 \
-			-fomit-frame-pointer -funroll-all-loops \
+			-Wall \
+			-funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
 
