CAN-2003-0865 fix taken from Conectiva Linux
--- mpg123-0.59r/httpget.c~	Fri May 28 08:10:44 1999
+++ mpg123-0.59r/httpget.c	Tue Nov 11 13:50:17 2003
@@ -53,11 +53,11 @@ void readstring (char *string, int maxle
 #if 0
 	char *result;
 #endif
 	int pos = 0;
 
-	while(1) {
+	while(pos < maxlen) {
 		if( read(fileno(f),string+pos,1) == 1) {
 			pos++;
 			if(string[pos-1] == '\n') {
 				string[pos] = 0;
 				break;
@@ -66,10 +66,11 @@ void readstring (char *string, int maxle
 		else if(errno != EINTR) {
 			fprintf (stderr, "Error reading from socket or unexpected EOF.\n");
 			exit(1);
 		}
 	}
+	string[pos] = 0;
 #if 0
 	do {
 		result = fgets(string, maxlen, f);
 	} while (!result  && errno == EINTR);
 	if (!result) {
