--- configure.ac	2006-10-21 00:08:34.000000000 +0200
+++ configure.ac-new	2006-10-25 02:45:07.530865360 +0200
@@ -144,7 +144,12 @@
 PKG_CHECK_MODULES(ESOUND, esound, HAVE_ESOUND=yes, HAVE_ESOUND=no)
 
 # Check for SDL
-PKG_CHECK_MODULES(SDL, sdl, HAVE_SDL=yes, HAVE_SDL=no)
+AC_CHECK_LIB( [SDL], [SDL_OpenAudio],
+		[ AC_CHECK_HEADER( [SDL/SDL.h],
+		[ HAVE_SDL="yes" ],
+		[ AC_MSG_WARN([Found SDL library but NOT header files on your system]) ] )
+	]
+)
 
 # Check for NAS
 AC_CHECK_LIB( [audio], [AuOpenServer],
--- src/audio_sdl.c	2006-10-21 00:08:33.000000000 +0200
+++ src/audio_sdl.c-new	2006-10-25 02:46:06.681873040 +0200
@@ -10,7 +10,7 @@
 #include <stdio.h>
 #include <math.h>
 
-#include <SDL.h>
+#include <SDL/SDL.h>
 
 #include "config.h"
 #include "debug.h"
