diff -ur mpg123-1.4.3.orig/Makefile.am mpg123-1.4.3/Makefile.am
--- mpg123-1.4.3.orig/Makefile.am	2008-05-20 20:50:32.000000000 +0200
+++ mpg123-1.4.3/Makefile.am	2008-05-29 21:35:30.000000000 +0200
@@ -7,8 +7,8 @@
 
 # libltdl first as this recursive make mess cannot resolve interdependencies
 # Why is recursive make the standard procedure when MAKE DOES NOT WORK THAT WAY??!
-SUBDIRS = @LTDLDIR@ src doc
-DIST_SUBDIRS = libltdl src doc
+SUBDIRS = src doc
+DIST_SUBDIRS = src doc
 
 # pkg-config file for the mpg123 library
 pkgconfigdir = $(libdir)/pkgconfig
