--- mpg123-0.59r/Makefile.wiget	Tue Oct 19 13:04:00 1999
+++ mpg123-0.59r/Makefile	Tue Oct 19 13:13:34 1999
@@ -14,17 +14,17 @@
 CFLAGS+=-DINET6
 
 ### KAME stack
-IPV6LIBDIR=-L/usr/local/v6/lib
-IPV6LIB=-linet6
-IPV6STACK="KAME"
+#IPV6LIBDIR=-L/usr/local/v6/lib
+#IPV6LIB=-linet6
+#IPV6STACK="KAME"
 ### BSD/OS 4.0 (NRL) stack
 #IPV6LIBDIR=
 #IPV6LIB=
 #IPV6STACK="BSD/OS 4.0"
 ### Linux stack
-#IPV6LIBDIR= #/usr/inet6/lib
-#IPV6LIB= #-linet6
-#IPV6STACK="Linux"
+IPV6LIBDIR= 
+IPV6LIB= 
+IPV6STACK="Linux"
 
 ###################################################
 ######                                       ######
@@ -127,7 +127,7 @@
 			audio_oss.o term.o' \
 		CFLAGS='$(CFLAGS) -DI386_ASSEM -DPENTIUM_OPT -DREAL_IS_FLOAT -DLINUX \
 			-DREAD_MMAP -DOSS -DTERM_CONTROL\
-			-Wall -O2 -m486 \
+			-Wall $(OPT_FLAGS) \
 			-fomit-frame-pointer -funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
@@ -138,7 +138,7 @@
 			decode_3dnow.o audio_oss.o term.o' \
 		CFLAGS='$(CFLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DLINUX \
 			-DUSE_3DNOW -DREAD_MMAP -DOSS -DTERM_CONTROL\
-			-Wall -O2 -m486 \
+			-Wall $(OPT_FLAGS) \
 			-fomit-frame-pointer -funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
@@ -149,7 +149,7 @@
 			decode_i486.o dct64_i486.o audio_oss.o term.o' \
 		CFLAGS='$(CFLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DI486_OPT -DLINUX \
 			-DREAD_MMAP -DOSS -DTERM_CONTROL\
-			-Wall -O2 -m486 \
+			-Wall $(OPT_FLAGS) \
 			-fomit-frame-pointer -funroll-all-loops \
 			-finline-functions -ffast-math' \
 		mpg123-make
@@ -161,20 +161,20 @@
 			audio_esd.o' \
 		CFLAGS='$(CFLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DLINUX \
 			-DREAD_MMAP -DOSS -DUSE_ESD \
-			-Wall  -O2 -m486 \
+			-Wall  $(OPT_FLAGS) \
 			-fomit-frame-pointer -funroll-all-loops \
 			-finline-functions -ffast-math \
 			$(RPM_OPT_FLAGS)' \
 		mpg123-make
 
 linux-alsa:
-	$(MAKE) CC=gcc LDFLAGS= \
+	$(MAKE) CC=gcc LDFLAGS='' \
 		AUDIO_LIB='-lasound' \
 		OBJECTS='decode_i386.o dct64_i386.o decode_i586.o \
 			audio_alsa.o term.o' \
 		CFLAGS='$(CFLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DLINUX \
 			-DREAD_MMAP -DALSA -DTERM_CONTROL\
-			-Wall  -O2 -m486 \
+			-Wall  $(OPT_FLAGS) \
 			-fomit-frame-pointer -funroll-all-loops \
 			-finline-functions -ffast-math \
 			$(RPM_OPT_FLAGS)' \
@@ -185,30 +185,30 @@
 		AUDIO_LIB='-lasound' \
 		OBJECTS='decode.o dct64.o audio_alsa.o term.o' \
 		CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DLINUX -DREAD_MMAP -DALSA \
-			-DTERM_CONTROL -Wall  -O2 \
+			-DTERM_CONTROL -Wall  \
 			-fomit-frame-pointer -funroll-all-loops \
 			-finline-functions -ffast-math \
-			$(RPM_OPT_FLAGS)' \
+			$(OPT_FLAGS)' \
 		mpg123-make
 
 linux-alpha:
 	$(MAKE) CC=gcc LDFLAGS= OBJECTS='decode.o dct64.o audio_oss.o' \
-		CFLAGS='$(CFLAGS) -DLINUX -DOSS -Wall -O2 \
+		CFLAGS='$(CFLAGS) -DLINUX -DOSS -Wall \
 			-fomit-frame-pointer -funroll-all-loops \
 			-finline-functions -ffast-math \
-			-Wall -O6 -DUSE_MMAP \
-			$(RPM_OPT_FLAGS)' \
+			-Wall -DUSE_MMAP \
+			$(OPT_FLAGS)' \
 		mpg123-make
 
 linux-alpha-esd:
 	$(MAKE) CC=gcc LDFLAGS= \
 		AUDIO_LIB='-lesd -laudiofile' \
 		OBJECTS='decode.o dct64.o audio_esd.o' \
-		CFLAGS='$(CFLAGS) -DLINUX -DOSS -Wall -O2 \
+		CFLAGS='$(CFLAGS) -DLINUX -DOSS -Wall \
 			-fomit-frame-pointer -funroll-all-loops \
 			-finline-functions -ffast-math \
-			-Wall -O6 -DUSE_MMAP \
-			$(RPM_OPT_FLAGS)' \
+			-Wall -DUSE_MMAP \
+			$(OPT_FLAGS)' \
 		mpg123-make
 
 #linux-ppc:
@@ -233,36 +233,40 @@
 linux-ppc:
 	$(MAKE) CC=gcc  LDFLAGS= \
 		OBJECTS='decode.o dct64.o audio_oss.o' \
-		CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DLINUX -Wall -O2 -mcpu=ppc \
+		CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DLINUX -Wall \
 			-DOSS \
 			-fomit-frame-pointer -funroll-all-loops \
-			-finline-functions -ffast-math' \
+			-finline-functions -ffast-math \
+			$(OPT_FLAGS)' \
 		mpg123-make
 
 linux-ppc-esd:
 	$(MAKE) CC=gcc  LDFLAGS= \
 		AUDIO_LIB='-lesd -laudiofile' \
 		OBJECTS='decode.o dct64.o audio_esd.o' \
-		CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DLINUX -Wall -O2 -mcpu=ppc \
+		CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DLINUX -Wall \
 			-DOSS  \
 			-fomit-frame-pointer -funroll-all-loops \
-			-finline-functions -ffast-math' \
+			-finline-functions -ffast-math \
+			$(OPT_FLAGS)' \
 		mpg123-make
 
 linux-sparc:
 	$(MAKE) CC=gcc  LDFLAGS= \
 		OBJECTS='decode.o dct64.o audio_sun.o' \
-		CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DUSE_MMAP -DSPARCLINUX -Wall -O2 \
+		CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DUSE_MMAP -DSPARCLINUX -Wall \
 			-fomit-frame-pointer -funroll-all-loops \
-			-finline-functions -ffast-math' \
+			-finline-functions -ffast-math \
+			$(OPT_FLAGS)' \
 		mpg123-make
 
 linux-m68k:
 	$(MAKE) CC=gcc LDFLAGS= OBJECTS='decode.o dct64.o audio_oss.o' \
 		CFLAGS='$(CFLAGS) -DREAL_IS_FLOAT -DLINUX -DREAD_MMAP \
-			-DOSS -DOSS_BIG_ENDIAN -Wall -O2 -m68040 \
+			-DOSS -DOSS_BIG_ENDIAN -Wall \
 			-fomit-frame-pointer -funroll-loops \
-			-finline-functions -ffast-math' \
+			-finline-functions -ffast-math \
+			$(OPT_FLAGS)' \
 		mpg123-make
 
 linux-sajber:
@@ -282,9 +286,10 @@
 		OBJECTS='decode_i386.o dct64_i386.o decode_i586.o \
 			control_sajber.o control_tk3play.o audio_oss.o' \
 		CFLAGS='$(CFLAGS) -DFRONTEND -DOSS -DI386_ASSEM -DREAL_IS_FLOAT \
-			-DPENTIUM_OPT -DLINUX -Wall -O2 -m486 \
+			-DPENTIUM_OPT -DLINUX -Wall \
 			-fomit-frame-pointer -funroll-all-loops \
-			-finline-functions -ffast-math' \
+			-finline-functions -ffast-math \
+			$(OPT_FLAGS)' \
 		$(FRONTEND)
 
 linux-nas:
@@ -293,9 +298,10 @@
 		OBJECTS='decode_i386.o dct64_i386.o audio_nas.o' \
 		CFLAGS='$(CFLAGS) -I/usr/X11R6/include \
 			-DI386_ASSEM -DREAL_IS_FLOAT -DLINUX -DNAS \
-			-Wall -O2 -m486 \
+			-Wall \
 			-fomit-frame-pointer -funroll-all-loops \
-			-finline-functions -ffast-math' \
+			-finline-functions -ffast-math \
+			$(OPT_FLAGS)' \
 		mpg123-make
 
 #### the following defines are for experimental use ... 
@@ -396,7 +402,7 @@
 			audio_oss.o' \
 		CFLAGS='$(CFLAGS) -DI386_ASSEM -DREAL_IS_FLOAT -DPENTIUM_OPT -DUSE_MMAP \
 			-DREAD_MMAP -DOSS \
-			-Wall -O2 -m486 \
+			-Wall $(RPM_OPT_FLAGS) \
 			-funroll-all-loops -finline-functions' \
 		mpg123-make
 
@@ -557,7 +563,7 @@
 # maybe you need the additonal options LDFLAGS='-lnsl -lsocket' when linking (see solaris:)
 generic:
 	$(MAKE) LDFLAGS= OBJECTS='decode.o dct64.o audio_dummy.o' \
-		CFLAGS='$(CFLAGS) -O -DGENERIC -DNOXFERMEM' \
+		CFLAGS='$(CFLAGS) -O -DGENERIC -DNOXFERMEM $(OPT_FLAGS)' \
 		mpg123-make
 
 ###########################################################################
