--- src/audio_nas.c	2006-10-24 23:56:58.000000000 +0200
+++ src/audio_nas.c-new	2007-01-02 01:10:49.000000000 +0100
@@ -224,10 +224,12 @@
         return -1;
 
     if (!(info.aud = AuOpenServer(ai->device, 0, NULL, 0, NULL, NULL))) {
-        if (ai->device==NULL)
+        if (ai->device==NULL) {
             error("could not open default NAS server");
-        else
+	}
+        else {
             error1("could not open NAS server %s\n", ai->device);
+	}
         return -1;
     }
     info.buf_size = 0;
