--- mpg123-1.5.0/configure.ac.orig	2008-04-29 22:15:44.000000000 +0200
+++ mpg123-1.5.0/configure.ac	2008-05-29 20:34:21.000000000 +0200
@@ -12,7 +12,7 @@
 
 AC_INIT([mpg123], [1.5.1], [mpg123-devel@lists.sourceforge.net])
 LIBMPG123_VERSION=7:0:7
-AC_SUBST( LIBMPG123_VERSION )
+AC_SUBST(LIBMPG123_VERSION)
 
 AC_CONFIG_SRCDIR(src/mpg123.c)
 AC_CONFIG_AUX_DIR(build)
@@ -650,8 +650,8 @@
 	echo $DECODER_LOBJ | grep -q $i.lo || DECODER_LOBJ="$DECODER_LOBJ $i.lo"
 done
 
-AC_SUBST( DECODER_OBJ )
-AC_SUBST( DECODER_LOBJ )
+AC_SUBST(DECODER_OBJ)
+AC_SUBST(DECODER_LOBJ)
 
 
 dnl ############## Output module choice
@@ -1109,11 +1109,11 @@
 AC_DEFINE_UNQUOTED( DEFAULT_OUTPUT_MODULE, "$default_output_modules", [The default audio output module(s) to use] )
 
 
-AC_SUBST( OUTPUT_OBJ )
-AC_SUBST( OUTPUT_MOD )
-AC_SUBST( OUTPUT_CFLAGS )
-AC_SUBST( OUTPUT_LIBS )
-AC_SUBST( OUTPUT_LDFLAGS )
+AC_SUBST(OUTPUT_OBJ)
+AC_SUBST(OUTPUT_MOD)
+AC_SUBST(OUTPUT_CFLAGS)
+AC_SUBST(OUTPUT_LIBS)
+AC_SUBST(OUTPUT_LDFLAGS)
 
 dnl ############## Compiler Optimizations
 
