diff -Nur mpg123.orig/decode_3dnow.s mpg123/decode_3dnow.s
--- mpg123.orig/decode_3dnow.s	Wed Oct 25 13:05:23 2000
+++ mpg123/decode_3dnow.s	Wed Feb  5 19:52:51 2003
@@ -55,7 +55,7 @@
 	femms
 		
 	# fixed by Takuhiro
-	cmpl $0,param+348
+	cmpl $0,param_equalfile
 	je .L25
 	pushl %ebx
 	pushl 48(%esp)
diff -Nur mpg123.orig/equalizer_3dnow.s mpg123/equalizer_3dnow.s
--- mpg123.orig/equalizer_3dnow.s	Wed Oct 25 13:05:23 2000
+++ mpg123/equalizer_3dnow.s	Wed Feb  5 19:52:42 2003
@@ -12,7 +12,7 @@
 	pushl %esi
 	pushl %ebx
 	movl 12(%esp),%ebx	# bandPtr
-	cmpl $0,param+348
+	cmpl $0,param_equalfile
 	je .L5
 	movl 16(%esp),%ecx	# channel
 	xorl %edx,%edx
diff -Nur mpg123.orig/mpg123.c mpg123/mpg123.c
--- mpg123.orig/mpg123.c	Thu Jan 18 15:00:35 2001
+++ mpg123/mpg123.c	Wed Feb  5 19:52:33 2003
@@ -69,7 +69,6 @@
     FALSE,  /* try to run process in 'realtime mode' */   
     { 0,},  /* wav,cdr,au Filename */
     NULL,   /* esdserver */
-    NULL,   /* equalfile */
     0,      /* enable_equalizer */
     32768,  /* outscale */
     0,      /* startFrame */
@@ -77,6 +76,7 @@
     0,      /* print_version:1 */
 };
 
+char *param_equalfile = NULL;
 
 static long numframes = -1;
 
@@ -318,7 +318,7 @@
     /* 'z' comes from the the german word 'zufall' (eng: random) */
     {'z', "shuffle",     0,                  0, &param.shuffle,    1},
     {'Z', "random",      0,                  0, &param.shuffle,    2},
-    {'E', "equalizer",	 GLO_ARG | GLO_CHAR, 0, &param.equalfile,1},
+    {'E', "equalizer",	 GLO_ARG | GLO_CHAR, 0, &param_equalfile,1},
     {0,   "aggressive",	 0,   	             0, &param.aggressive,2},
 #ifdef USE_3DNOW
     {0,   "force-3dnow", 0,                  0, &param.stat_3dnow,1},
@@ -736,7 +736,7 @@
 
     audio_capabilities(&ai);
 
-    if(param.equalfile) { /* tst */
+    if(param_equalfile) { /* tst */
 	FILE *fe;
 	int i;
 
@@ -746,7 +746,7 @@
 	    equalizer_sum[0][i] = equalizer_sum[1][i] = 0.0;
 	}
 
-	fe = fopen(param.equalfile,"r");
+	fe = fopen(param_equalfile,"r");
 	if(fe) {
 	    char line[256];
 	    for(i=0;i<32;i++) {
@@ -766,7 +766,7 @@
 	    param.enable_equalizer = 1;
 	}
 	else
-	    fprintf(stderr,"Can't open equalizer file '%s'\n",param.equalfile);
+	    fprintf(stderr,"Can't open equalizer file '%s'\n",param_equalfile);
     }
 
 #if !defined(WIN32) && !defined(GENERIC) && !defined(MINT) && !defined(__EMX__) && !defined(OS2)
diff -Nur mpg123.orig/mpg123.h mpg123/mpg123.h
--- mpg123.orig/mpg123.h	Tue Jan 23 18:58:28 2001
+++ mpg123/mpg123.h	Wed Feb  5 19:52:00 2003
@@ -216,7 +216,6 @@
     int realtime;
     char filename[256];
     char *esdserver;
-    char *equalfile;
     int  enable_equalizer;
     long outscale;
     long startFrame;
@@ -450,6 +449,7 @@
 extern struct audio_name audio_val2name[];
 
 extern struct parameter param;
+extern char *param_equalfile;
 
 /* 486 optimizations */
 #define FIR_BUFFER_SIZE  128
--- mpg123.orig/audio_esd.c	Mon May 22 11:45:14 2000
+++ mpg123/audio_esd.c	Wed Feb  5 20:33:51 2003
@@ -21,7 +21,7 @@
     esd_server_info_t *info;
     esd_format_t fmt;
 
-    if ((esd = esd_open_sound(NULL)) >= 0)
+    if ((esd = esd_open_sound(param.esdserver)) >= 0)
     {
       info = esd_get_server_info(esd);
       esd_rate = info->rate;
