--- mpg123-1.23.8/src/libout123/modules/alsa.c.orig	2016-09-26 23:20:25.000000000 +0200
+++ mpg123-1.23.8/src/libout123/modules/alsa.c	2016-10-08 13:31:57.348509990 +0200
@@ -13,7 +13,7 @@
      http://mailman.alsa-project.org/pipermail/alsa-devel/2007-June/001684.html
    ... seems like the conclusion was not carried through.
  */
-#define _POSIX_SOURCE
+#define _POSIX_C_SOURCE 199309L /* for struct timespec in time.h */
 #include "out123_int.h"
 #include <errno.h>
 
